common/txmgr/confirmer.go

```diff
@@ -858,10 +858,19 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) han
 		}
 		return ec.handleInProgressAttempt(ctx, lggr, etx, replacementAttempt, blockHeight)
 	case client.ExceedsMaxFee:
-		// Confirmer: The gas price was bumped too high. This transaction attempt cannot be accepted.
-		// Best thing we can do is to re-send the previous attempt at the old
-		// price and discard this bumped version.
-		fallthrough
+		// Confirmer: Note it is not guaranteed that all nodes share the same tx fee cap.
+		// So it is very likely that this attempt was successful on another node since
+		// it was already successfully broadcasted. So we assume it is successful and
+		// warn the operator that the RPC node is misconfigured.
+		// This failure scenario is a strong indication that the RPC node
+		// is misconfigured. This is a critical error and should be resolved by the
+		// node operator.
+		// If there is only one RPC node, or all RPC nodes have the same
+		// configured cap, this transaction will get stuck and keep repeating
+		// forever until the issue is resolved.
+		lggr.Criticalw(`RPC node rejected this tx as outside Fee Cap but it implies that it was successful on another node previously`, "attempt", attempt)
```

Comment:
This description is incorrect. There is no guarantee the tx got accepted by some other node, or that we had any successful tx to begin with. For all we know the attempt we are trying to send in this execution path could be a bumped attempt, which is indeed higher than the configured cap. If we store this attempt we run the risk of getting stuck, instead of resending the previous attempt, which potentially was within the limits.

Reply:
But there's a chance that this TxAttempt got accepted by some node. And there's a chance that this TxAttempt will make it onchain. So we need to save and track this TxAttempt for receipts.

>>If we store this attempt we run the risk of getting stuck, instead of resending the previous attempt, which potentially was within the limits.
We don't run the risk of getting stuck, because we will still be resending both these attempts always. Any 1 of these attempts can get confirmed.

Reply:
AFAIK Resender & Confirmer both send or bump the highest-priced attempt stored so far, not all the in-between attempts. That's why we get stuck on caps so ofter. This means if we store this one and it's not priced correctly you  enter a state where the attempt you have can never be accepted.

Reply:
And as for the comment, I wasn't implying that it can't happen, I just pointed out that the description was misleading on how often it could happen and what are the assumptions.

Reply:
I think the Resender resends all the attempts. We should check and confirm that.
But even if it doesn't resend older attempts, there's no reason to believe the MaxPriced ones aren't getting accepted.

Reply:
`// FindTxAttemptsRequiringResend returns the highest priced attempt for each eth_tx that was last sent before or at the given time (up to limit)`
https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/txmgr/evm_tx_store.go#L707

> But even if it doesn't resend older attempts, there's no reason to believe the MaxPriced ones aren't getting accepted.

I'm not saying it can't happen, I'm pointing out that either of these things can be true:  some other node accepted the tx **or** no nodes accepted the tx. The comment and the fix here assume the first case is always true.

Reply:
If the 1st case is true, then we are correctly bumping and tracking that attempt.
If the 2nd case is true, then we really cannot do anything till the NOP fixes their node.

If not this fix, what is the other suggestion?